peps/pep-0723.rst

```diff
@@ -101,14 +102,38 @@ This PEP defines a metadata comment block format loosely inspired [2]_ by
 
 __ https://docutils.sourceforge.io/docs/ref/rst/directives.html
 
-Any Python script may have top-level comment blocks that start with the line
-``# /// TYPE`` where ``TYPE`` determines how to process the content, and ends
-with the line ``# ///``. Every line between these two lines MUST be a comment
-starting with ``#``. If there are characters after the ``#`` then the first
-character MUST be a space. The embedded content is formed by taking away the
-first two characters of each line if the second character is a space, otherwise
-just the first character (which means the line consists of only a single
-``#``).
+Any Python script may have top-level comment blocks that MUST start with the
+line ``# /// TYPE`` where ``TYPE`` determines how to process the content. That
+is: a single ``#``, followed by a single space, followed by three forward
+slashes, followed by a single space, followed by the type of metadata. Block
+MUST end with the line ``# ///``. That is: a single ``#``, followed by a single
+space, followed by three forward slashes. The ``TYPE`` MUST only consist of
+ASCII letters, numbers and hyphens.
+
+Every line between these two lines (``# /// TYPE`` and ``# ///``) MUST be a
+comment starting with ``#``. If there are characters after the ``#`` then the
+first character MUST be a space. The embedded content is formed by taking away
+the first two characters of each line if the second character is a space,
+otherwise just the first character (which means the line consists of only a
+single ``#``).
+
+Precedence for an ending line ``# ///`` is given when the next line is not
+a comment. For example, the following is a single fully valid block:
```

Comment:
[As mentioned in the discuss thread](https://discuss.python.org/t/pep-722-723-decision/36763/92?u=davidcellis) this is not quite as strict as the regex.

eg:
```
# /// pyproject
# test value
# ///
##
# /// metadata
# more data
# ///
```

`##` is a comment, but this is parsed as 2 blocks because it is not a valid embedded content line (for lack of a better term).

I think it would more accurately be something like this:
```suggestion
Precedence for an ending line ``# ///`` is given when the next line is not
a valid embedded content line as described above. 
For example, the following is a single fully valid block:
```

Someone else may have better language though.

Reply:
Thanks!